### PR TITLE
Change to binary prefixes in progress.py

### PR DESCRIPTION
**Pull request by amartya-dev**
## Type of changes

- [ ] Bug fix
- [ ] New feature
- [ ] Documentation / docstrings
- [ ] Tests
- [x] Other

## Checklist

- [x] I've run the latest [black](https://github.com/psf/black) with default args on new code.
- [ ] I've updated CHANGELOG.md and CONTRIBUTORS.md where appropriate.
- [ ] I've added tests for new code.
- [x] I accept that @willmcgugan may be pedantic in the code review.

## Description

Incorporates #323 

I have just changed the prefixes to binary as suggested in the issue

Edit: Checked off a few more boxes and added the description at the correct place

### CODE REVIEW

File: rich/progress.py [hunk 1]
```diff
@@ -299,7 +299,9 @@ def render(self, task: "Task") -> Text:
         completed = int(task.completed)
         total = int(task.total)
         unit, suffix = filesize.pick_unit_and_suffix(
-            total, ["bytes", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB"], 1024
+            total,
```
Comment (willmcgugan) [inline rich/progress.py:302]: I think filesize should be in decimal units, to be consistent with file managers. i.e. same units, just with the third parameter as 1000.

**Comment by willmcgugan** [comment]
Great. Please add a test for this and an entry in the README.

File: rich/progress.py (hunk 1 above)
Comment (amartya-dev) [reply rich/progress.py:302]: Sure let me change that

**Review by willmcgugan — CHANGES_REQUESTED** [review]

File: tests/test_progress.py [hunk 2]
```diff
@@ -254,6 +254,40 @@ def test_columns() -> None:
     assert result == expected
 
 
+def test_download_progress_uses_decimal_units() -> None:
```
[the remark this answers is not in the document]
Comment (amartya-dev) [reply tests/test_progress.py:257]: Ok, I will look into it...

File: CHANGELOG.md [hunk 3]
```diff
@@ -19,6 +19,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Changed Style.empty to Style.null to better reflect what it does
 - Optimized combining styles involving a null style
 - Change error messages in Style.parse to read better
+- Change the render prefix to correspond to the decimal units in progress
```
[the remark this answers is not in the document]
Comment (amartya-dev) [reply CHANGELOG.md:22]: Ok

[the remark this answers is not in the document]
Comment (amartya-dev) [reply CHANGELOG.md:22]: What date should I use there?

[the remark this answers is not in the document]
Comment (willmcgugan) [reply CHANGELOG.md:22]: Just say "unreleased" for now. https://semver.org/spec/v2.0.0 is a worthwhile read.

**Comment by amartya-dev** [comment]
@willmcgugan could you please check and let me know if any further changes are required?

File: tests/test_progress.py [hunk 4]
```diff
@@ -79,6 +79,28 @@ class FakeTask(Task):
     assert str(text) == "0:01:00"
 
 
+def test_download_progress_uses_decimal_units() -> None:
```
Comment (willmcgugan) [inline tests/test_progress.py:82]: Better, but too complex for a test.

I see you're trying to be thorough (which is good!) but a test should be no more complex that is needed to catch a bug.

Have a look at `test_time_remaining_column`, you'll notice that there are no loops, just assignments and method calls, but that's enough to execute every line of the changed code. I think you can do something similar here...

**Review by willmcgugan — CHANGES_REQUESTED** [review]

File: tests/test_progress.py (hunk 4 above)
Comment (amartya-dev) [reply tests/test_progress.py:82]: Ok, makes sense, I thought I will add test all the symbols. Should I just create a task say half completed and just assert the rendered string?

Comment (willmcgugan) [reply tests/test_progress.py:82]: Sounds reasonable 👍

**Comment by amartya-dev** [comment]
@willmcgugan are there any further changes required?

**Comment by willmcgugan** [comment]
LGTM Thanks.

**Pull request merged**